src/api-service/__app__/onefuzzlib/azure/image.py

```diff
@@ -21,12 +21,20 @@ def get_os(region: Region, image: str) -> Union[Error, OS]:
     client = get_compute_client()
     parsed = parse_resource_id(image)
     if "resource_group" in parsed:
-        try:
-            name = client.images.get(
-                parsed["resource_group"], parsed["name"]
-            ).storage_profile.os_disk.os_type.name
-        except (ResourceNotFoundError, CloudError) as err:
-            return Error(code=ErrorCode.INVALID_IMAGE, errors=[str(err)])
+        if parsed["type"] == "galleries":
+            try:
+                name = client.gallery_images.get(
+                    parsed["resource_group"], parsed["name"], parsed["child_name_1"]
+                ).os_type.lower()
+            except (ResourceNotFoundError, CloudError) as err:
+                return Error(code=ErrorCode.INVALID_IMAGE, errors=[str(err)])
+        else:
+            try:
+                name = client.images.get(
+                    parsed["resource_group"], parsed["name"]
+                ).storage_profile.os_disk.os_type.lower()
+            except (ResourceNotFoundError, CloudError) as err:
```

Comment:
I get the motivation, but I'm not sure it really makes sense to scope these `except` blocks to only these Azure client calls. Not an issue for this PR, though.